ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1442,7 +1475,7 @@ void checkAndMaybeComplete() {
       // Now, check if this chunk is done.
       if (operationTracker.isDone() || (chunkException != null
           && chunkException.getErrorCode() == RouterErrorCode.TooManyRequests)) {
-        if (!operationTracker.hasSucceeded()) {
+        if (!operationTracker.hasSucceeded() && !shouldUseReservedMetadataId()) {
```

Comment:
Do we have an issue here?
The else condition of if (!operationTracker.hasSucceeded) will set the "chunkException" to null and "done to true".
That means if it's meta chunk failure, it'll be treated as successfully.

I guess your intention is to skip failure retry. If so, we should add the "!shouldUseReservedMetadataId" condition to the following 
```
if (failedAttempts <= routerConfig.routerMaxSlippedPutAttempts) {
}
```

Reply:
Do you have the test case to cover metadata write failure?

Reply:
Thats right. Thanks for catching it. Fixed it.